cli/crates/server/src/custom_resolvers.rs

```diff
@@ -0,0 +1,217 @@
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::process::Stdio;
+
+use common::environment::Environment;
+use futures_util::{pin_mut, TryStreamExt};
+use tokio::process::Command;
+
+use crate::errors::ServerError;
+
+#[derive(strum_macros::AsRefStr)]
+enum CommandType {
+    Npm,
+    Npx,
+}
+
+async fn run_npm_command<P: AsRef<Path>>(
+    command_type: CommandType,
+    artifact_directory_path: P,
+    extra_arguments: &[&str],
+    tracing: bool,
+    environment: &[(&'static str, &'static str)],
+) -> Result<(), ServerError> {
+    use itertools::Itertools;
+
+    let artifact_directory_path_string = artifact_directory_path
+        .as_ref()
+        .to_str()
+        .ok_or(ServerError::CachePath)?;
+
+    let mut arguments = vec!["--prefix", artifact_directory_path_string];
+    arguments.extend(extra_arguments);
+
+    trace!("running 'npm {}'", arguments.iter().format(" "));
```

Comment:
Should this use `CommandType` rather than assuming it's always `npm`?

Reply:
Definitely. Fixed.